hal/build.rs

```diff
@@ -0,0 +1,21 @@
+use std::env;
+
+fn main() -> Result<(), &'static str> {
+    // Refer to
+    // <https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts>
+    fn feat(f: &str) -> bool {
+        env::var(format!(
+            "CARGO_FEATURE_{}",
+            f.to_ascii_uppercase().replace("-", "_")
+        ))
+        .is_ok()
+    }
+
+    if !feat("device-selected") {
+        return Err(
+            "The HAL is built for a specific target device selected using a feature, but no such a feature was selected."
+        );
+    }
+
```

Comment:
Checks for VDDIO are missing. A fail must also be added for ` (v70 || v71) && vddio-1v`.

Reply:
Thinking over it now, I believe VDDIO information should be provided to the EFC module instead of being set statically. Consider the case when the MCU is dynamically downvolted. Using the API for this information is also much stronger than an environmental variable.

Reply:
See https://github.com/atsams-rs/atsamx7x-rust/pull/16/commits/762e919bbf644113496becf1e15a2909a8410e4d.